lib/monitor/run.js

```diff
@@ -343,15 +364,21 @@ function kill(child, signal, callback) {
     const sig = signal.replace('SIG', '');
     psTree(child.pid, function (err, kids) {
       if (psTree.hasPS) {
-        spawn('kill', ['-s', sig, child.pid].concat(kids))
-          .on('close', callback);
+        spawn('kill', ['-s', sig].concat(kids))
+          .on('close', waitForOldKids.bind(this, kids, function () {
+            spawn('kill', ['-s', sig, child.pid])
+              .on('close', callback);
+          }));
       } else {
         // make sure we kill from smallest to largest
-        const pids = kids.concat(child.pid).sort();
```

Comment:
I'm assuming you removed the `concat` here because `psTree` already includes the given `child.pid` in the list.